prov/slurm/helper_test.go

```diff
@@ -350,3 +351,63 @@ func TestParseJob(t *testing.T) {
 	require.Equal(t, "test-salloc-Environment", info["JobName"], "unexpected value for \"JobName\" key")
 	require.Equal(t, "2-19:42:53", info["RunTime"], "unexpected value for \"RunTime\" key")
 }
+
+func TestGetJobInfoWithInvalidJob(t *testing.T) {
+	t.Parallel()
+	s := &MockSSHClient{
+		MockRunCommand: func(cmd string) (string, error) {
+			return "slurm_load_jobs error: Invalid job id specified", errors.New("")
+		},
+	}
+	info, err := getJobInfo(s, "1234")
+	require.Nil(t, info, "info should be nil")
+
+	require.Equal(t, true, isNoJobFoundError(err), "expected no job found error")
+}
+
+func TestGetJobInfo(t *testing.T) {
+	t.Parallel()
+	s := &MockSSHClient{
+		MockRunCommand: func(cmd string) (string, error) {
+			content, err := ioutil.ReadFile("testdata/scontrol.txt")
+			require.Nil(t, err, "Unexpected error reading scontrol")
+			return string(content), nil
+		},
+	}
+
+	data, err := os.Open("testdata/scontrol.txt")
+	require.Nil(t, err, "unexpected error while opening test file")
+	expected, err := parseJobInfo(data)
+	require.Nil(t, err, "Unexpected error parsing job info")
+	info, err := getJobInfo(s, "1234")
+	require.Nil(t, err, "Unexpected error retrieving job info")
+	require.NotNil(t, info, "info should not be nil")
+
+	require.Equal(t, expected, info, "unexpected job info")
+}
+
+func TestGetJobInfoWithEmptyResponse(t *testing.T) {
+	t.Parallel()
+	s := &MockSSHClient{
+		MockRunCommand: func(cmd string) (string, error) {
+			return "", nil
+		},
+	}
+	info, err := getJobInfo(s, "1234")
+	require.Nil(t, info, "info should be nil")
+
+	require.Equal(t, true, isNoJobFoundError(err), "expected no job found error")
+}
+
+func TestGetJobInfoWithError(t *testing.T) {
+	t.Parallel()
+	s := &MockSSHClient{
+		MockRunCommand: func(cmd string) (string, error) {
+			return "oups, it's bad", errors.New("this is an error !")
+		},
+	}
+	info, err := getJobInfo(s, "1234")
+	require.Nil(t, info, "info should be nil")
+
+	require.Equal(t, "oups, it's bad: this is an error !", err.Error(), "expected error")
```

Comment:
"oup"s is the french writing, should be "oops" :-)

Reply:
oops !